gui/aju_flask/aju_app/__init__.py

```diff
@@ -20,3 +37,48 @@ def create_app(config_name):
     app.register_blueprint(main_blueprint)
 
     return app
+
+
+@socketio.on('connect')
+def send_kafka_data():
+    global thread
+    with thread_lock:

```

Comment:
This thread will not close until the the server is down. Because this is a communication tunnel between the client and the server. So it will not be released.

Reply:
So how are you handling graceful shutdown?

Reply:
For now, there is not the clearance after the user closing the web page. We can add some clearance support in the future.